data/data/azure/variables-azure.tf

```diff
@@ -0,0 +1,51 @@
+variable "azure_config_version" {
+  description = <<EOF
+(internal) This declares the version of the Azure configuration variables.
+It has no impact on generated assets but declares the version contract of the configuration.
+EOF
+
+  default = "0.1"
+}
+
+variable "azure_region" {
+  type        = "string"
+  description = "The target Azure region for the cluster."
+}
+
+variable "azure_bootstrap_vm_type" {
+  type        = "string"
+  description = "Instance type for the bootstrap node. Example: `Standard_DS4_v3`."
+}
+
+variable "azure_master_vm_type" {
+  type        = "string"
+  description = "Instance type for the master node(s). Example: `Standard_DS4_v3`."
+}
+
+// This variable is generated by OpenShift internally. Do not modify
+variable "cluster_name" {
```

Comment:
why do you need this. AWS doesn't have this. https://github.com/openshift/installer/blob/5909a4881add13347c568b7a6c049466b6156f1f/data/data/aws/variables-aws.tf

Reply:
This is to satisfy the urls expected by the installer and ignition, and stay in the private vnet during ignition.
I had to create two DNS zone. One _public_, and one _private_.
You cannot create both with the same `cluster_domain` (`<cluster_name>.<domain_base>`), even though one is public and the other one is private.

The workaround is to use `cluster_domain` for the public one, and compose the private one with `base_domain` + a cname as `api.<cluster_name>`. that is where I need `cluster_name`.

This allows the internal network calls to stay in the vnet (redirected to the internal LB), while also exposing the public url as expected by the installer (pointing at the external LB)

I can remove it if we drop the public DNS and rely on the automatic `<label>.cloudapp.azure.com` one assigned by azure to the public IP. However, it would prevent you from setting up your own domain name as external endpoint.

I suggest to iterate over this after we have an initial working solution

Reply:
on AWS:
- we do not create public DNS zone, we consume the public DNS zone corresponding `basedomain` and create records in that Zone of  the form `api.$cluster_domain<$clsuter_name.$base_domain>`

- we create the private DNS zone for `$cluster_domain<$clsuter_name.$base_domain>` and create all necessary provate records there.

see [here](https://github.com/openshift/installer/blob/master/data/data/aws/route53/base.tf)

Hopefully

Reply:
> If we do that, does our approach match your expectations?
`our approach` regarding dns zone, or entire PR. coz i'm still looking at the PR haven't finished...

Reply:
@abhinavdahiya I'm asking about the DNS zone. Help us understand what the requirements are for the public DNS zone because "on AWS" does not suffice. @serbrech's change removes the need for a manual step of configuring DNS for the azure case. I'd like to know why this isn't desirable over manual steps.

Reply:
> @abhinavdahiya I'm asking about the DNS zone. Help us understand what the requirements are for the public DNS zone because "on AWS" does not suffice. @serbrech's change removes the need for a manual step of configuring DNS for the azure case. I'd like to know why this isn't desirable over manual steps.

My concern is on the requirement to bleed `clustername` across installer -> terraform boundary to go this automated direction. that affects the api installer exposes to terraform codebase.

Reply:
@serbrech @abhinavdahiya @juan-lee IIRC the historical reason for having to manually create the public zone is that you need to know ahead of time the name servers of the zone (e.g ns-*.awsdns-*.com) in order to configure your dns provider to point to them. Any automation to overcome the manual step sgtm but I'd be in favour of going for the simplest way for now so we have an minimum operational version

Reply:
https://github.com/MicrosoftDocs/azure-docs/issues/8249

Reply:
I also filed feedback internally, in the meantime, I'll follow your existing behaviour to list the found public DNS in the subscription during the installer survey, and fail if none is found.